### ISSUE TITLE
[Bug]: Unable to delete account as banned user

### ISSUE THREAD

**Issue by IzzyData** [LemmyNet/lemmy#3973]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

Having a banned account prevents that user from self deleting their account in settings due to not being able to log in. 

It might make sense to to allow that user to log in and take no actions other than modify their own settings and delete the account. 

### Steps to Reproduce

1. Ban a user
2. Attempt to log in

### Technical Details

This will be the case for all environments.

### Version

BE: 0.18.4

### Lemmy Instance URL

lemmy.ml lemmy.world

**Comment by DraconicNEO** [comment]
I would agree with that, people should be able to delete comments or posts they've made even when banned, as well as be able to purge their account if they so choose.
Though I don't really think changing settings is a good idea since they could still abuse that.

Here's what I think people should be allowed to do in a Banned account:
- Delete Posts and Comments
- Initiate Account Deletion
- Possibly Message Admins

Things they shouldn't be able to do:
- Vote
- Post
- Comment
- Message Users
- Upload Images
- Alter User Settings

Reddit has a similar feature with account suspension. When a user is suspended they are still able to log in but they cannot vote, comment, create posts, send messages to users (only to Admins). They can however delete posts and comments and also delete their account.

**Comment by IzzyData** [comment]
I agree with DraconicNEO's assessment. I've found this to be the way most similar sites to Lemmy work

**Comment by maltfield** [comment]
This should be high priority. Currently it means that lemmy instance admins that have banned any users in the EU are violating the "Right to Erasure" of the GDPR.

The fines for this violation are millions of euros or a percent of of revenue, whichever is higher.

This law applies to any website operating anywhere in the world (not just to websites or businesses located in the EU) that has users who are residents of the EU.

**Comment by maltfield** [comment]
See also [[Bug]: Delete Account should delete uploaded media (pictures) too (GDPR right to erasure violation)](https://github.com/LemmyNet/lemmy/issues/4433)

**Comment by DraconicNEO** [comment]
> See also [[Bug]: Delete Account should delete uploaded media (pictures) too (GDPR right to erasure violation)](https://github.com/LemmyNet/lemmy/issues/4433)

Isn't this issue fixed in 0.19.3 with their [Image upload log](https://github.com/LemmyNet/lemmy/pull/3927) that will delete all uploads after user deletes their account.

**Comment by maltfield** [comment]
@DraconicNEO you commented in the wrong ticket, but the answer is no.

Please continue any further discussion in the relevant ticket.

**Comment by dessalines** [comment]
@maltfield could you take on this issue? We're a little swamped at the moment.

**Comment by sunaurus** [comment]
I don't think this is strictly required by the GDPR. The GDPR allows retaining data for some purposes, it seems to me that preventing abuse would be a perfectly valid reason. If the GDPR would really allow banned users to delete their data, then bans would have no effect, as users could simply evade them with GDPR data deletion at any time, right?

**Comment by DraconicNEO** [comment]
> I don't think this is strictly required by the GDPR. The GDPR allows retaining data for some purposes, it seems to me that preventing abuse would be a perfectly valid reason. If the GDPR would really allow banned users to delete their data, then bans would have no effect, as users could simply evade them with GDPR data deletion at any time, right?
> 
> Edit: looked into it a bit more, found this on law.stackexchange.com which seems to support my understanding: https://law.stackexchange.com/a/37916

I mean when a user is banned we still keep the appropriate records on hand, their name is still in the bans list and so is their email and whatever other data is used. Posts and comments aren't part of that needed data, nor are their profile entries. 

IMO giving people the ability to login but not do anything other than Deleting or exporting basically kills any chance of them being able to use regulation against an instance, as it gives them the ability to do it then and there. It's likely why Reddit opted to have account suspensions that fully restrict the account as opposed to locking them out, and they have the money and time to deal with legal trolls, smaller Lemmy instances likely don't.

**Comment by maltfield** [comment]
@dessalines I'm not a rust developer, but I will try to contribute some better documentation to the lemmy project in relation to similar issues this month.

**Comment by maltfield** [comment]
For additional context of this issue, please see [Nightmare on Lemmy Street (A Fediverse GDPR Horror Story)](https://tech.michaelaltfield.net/2024/03/04/lemmy-fediverse-gdpr/)

[![Nightmare on Lemmy St - A GDPR Horror Story](https://tech.michaelaltfield.net/wp-content/uploads/sites/5/lemmy-fediverse-gdpr_featuredImage2.jpg)](https://tech.michaelaltfield.net/2024/03/04/lemmy-fediverse-gdpr/)

**Comment by Nothing4You** [comment]
no it's not, the api still requires a `LocalUserView`, which requires a valid JWT, which requires not being banned to obtain one. existing JWTs are invalidated when a user gets banned.

**Comment by Nothing4You** [comment]
given that a password is already required anyway, this should be fairly straightforward to implement though on the backend side by just decoupling it from the `LocalUserView` extractor and instead always authenticating the user with the given username (or email) and password.

**Comment by dessalines** [comment]
Yep, that should be easy. I don't know how front-ends would handle this tho, probably would need to be a special page.
